src/picterra/client.py

```diff
@@ -82,22 +83,38 @@ def _wait_until_operation_completes(self, operation_response):
                 raise APIError('Operation %s failed' % operation_id)
             time.sleep(poll_interval)
 
-    def _paginate_through_list(self, resource_endpoint: str):
+    def _paginate_through_list(self, resource_endpoint: str, params: dict = {}):
```

Comment:
Don't use [mutable as default values](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) in Python. The right way to write this is:
```
def _paginate_through_list(self, ..., params = None):
  if params is None:
    params = {}
```

Reply:
Checked